qa/die-with-dignity/build.gradle

```diff
@@ -27,11 +27,11 @@ esplugin {
 integTestRunner {
     systemProperty 'tests.security.manager', 'false'
     systemProperty 'tests.system_call_filter', 'false'
-    systemProperty 'pidfile', "${-> integTest.getNodes().get(0).pidFile}"
-    systemProperty 'log', "${-> integTest.getNodes().get(0).homeDir}/logs/${-> integTest.getNodes().get(0).clusterName}_server.json"
+    nonInputProperties.systemProperty 'pidfile', "${-> integTest.getNodes().get(0).pidFile}"
```

Comment:
Wouldn't this work without `nonInputProperties` due to the closure in the Gstring ? 
I think this extension is confusing and complicates builds scripts.

Reply:
All `GStrings` which have a closure that includes a reference to something non-serializable will blow up at input snapshotting time. In this case you can see this closure refers to `integTest` which is a `Task`, which is most definitely not serializable.

Essentially, we need a way to be able to pass a system property to tasks, but not have Gradle treat it as an input. We actually need this for two reasons. First for the reason mentioned above, which is that property values that cannot be serialized blow up input snapshotting. Second, is that some of these will be unstable in nature and will break cacheability, and we want to be able to tell Gradle to ignore them.

We can certainly tweak the DSL and make it better. Perhaps rather than `nonInputProperties.systemProperty` we just make it `nonInputSystemProperty`?

Reply:
Do you think something like `dyncamicSystemProperty`  that reveals more context for the test author would be more appropriate ? I'm concerned that for someone not that  intimate with the build and Gradle `nonInput` will be meaningless, or create a connotation that  these are not being sent to the test runner.

Also, while at it, we should only have it take a closure and get rid of all the GString hacks for clarity.

Reply:
> Do you think something like dyncamicSystemProperty that reveals more context for the test author would be more appropriate ?

Yeah, I'm not sure. Mainly because we will certainly use this for things that are not "dynamic", in the sense that we won't always pass a lazy G-string here. We'll use this for anything that changes and we want Gradle to ignore as an input, for example, for local and pr builds, the test seed itself.

> I'm concerned that for someone not that intimate with the build and Gradle nonInput will be meaningless

I'm sure this isn't the first case of developers interacting with something in the build that they don't fully grasp 😉. The goal should be to hide some of this complexity behind plugins so they aren't in build scripts. I think we should consider doing that instead as many of these instances are duplicates and could probably simply be part of the `RestIntegTestTask` (or a plugin we convert it to).

> Also, while at it, we should only have it take a closure and get rid of all the GString hacks for clarity.

Agreed. We can do that simply enough.

Reply:
How about `notCachable` or `noCache` instead of `nonInput` ? 
I find `nonInput` really confusing as the system properties _are_inputs to the test and there's no indication that this refer to task.inputs

Reply:
@rjernst care to join the bikeshedding? 😄

Reply:
I'm fine with `nonInput` for now. I think it is something we can discuss further if there is confusion and rename easily.